rclcpp/src/rclcpp/utilities.cpp

```diff
@@ -191,6 +191,54 @@ rclcpp::init(int argc, char * argv[])
 #endif
 }
 
+std::vector<std::string>
+rclcpp::remove_ros_arguments(int argc, char const * const argv[])
+{
+  rcl_allocator_t alloc = rcl_get_default_allocator();
+  rcl_arguments_t parsed_args;
+
+  if (rcl_parse_arguments(argc, argv, alloc, &parsed_args) != RCL_RET_OK) {
```

Comment:
@sloretz was there going to be a function to get the already parsed arguments?

`rcl_init` already does this work and keeps the `rcl_arguments_t` around for later use. So we could use that here. The downside is that this would need `rcl_init` to have been called before, which might actually be a reason to leave this as-is.

Also, we'd need to ensure the `argc` and `argv` are the same ones given to the previously called `rcl_init`.

So with those two caveat's maybe this is the best approach.

Reply:
@wjwwood You mean a function that gets the global parsed arguments? There is [`rcl_get_global_arguments`](https://github.com/ros2/rcl/blob/e48a445f46f1da21ff0c4bf3d52bec9120ab7b57/rcl/include/rcl/arguments.h#L181). Both your caveats apply.

Reply:
Oh yeah, forgot you ended up adding that. Any opinion on how this function should work, i.e. should it reused the result of init or do it again itself to avoid the corner cases?

Reply:
As a standalone function with `argc` and `argv` passed in +1 to parsing the args again to prevent a mismatch between the arguments and the global state.

A non-public `rclcpp` function that made an `std::vector` of non-ros arguments given `argc`, `argv`, and `rcl_arguments_t` could be used by both `remove_ros_arguments` and  `init_and_remove_ros_arguments` to avoid the double-parsing. However, inefficiency in `init..()` is a one-time cost in most use cases so it may not be worth the added work.

Reply:
Sounds good to me too. @mjcarroll I'll leave it up to you to either use the common function or not as @sloretz described.

Reply:
This issue would be OBE if we collapse init into a single signature that always returns a std::vector.